### PR TITLE
fix pesq docs

### LINKED ISSUE #2095: Inconsistent docs and code output for `PerceptualEvaluationSpeechQuality `
## 🐛 Bug
[related to #1092]

The [docs](https://torchmetrics.readthedocs.io/en/stable/audio/perceptual_evaluation_speech_quality.html) for PESQ metric class state that 
> As output of _forward_ and _compute_ the metric returns the following output `pesq` (Tensor): float tensor with shape (...,) of PESQ value per sample.

However, I always get one value, no matter the `batch_size`, as if the metric was being averaged.

### To Reproduce

```python
import torch 
from torchmetrics.audio import PerceptualEvaluationSpeechQuality as PESQ

pesq = PESQ(fs=16000, mode="wb")

# create data
batch_size = 4
audio_size = 12345
preds = torch.FloatTensor(batch_size, audio_size).uniform_(-1, 1)
target = torch.FloatTensor(batch_size, audio_size).uniform_(-1, 1)
print("preds.shape:", preds.shape)
#>preds.shape: torch.Size([4, 12345])

# compute metric
pesq_score = pesq.forward(preds, target)
print("pesq_score:", pesq_score)
#>pesq_score: tensor(1.5049)
```
I expected the output to be of shape `torch.Size([4])`.
Same behaviour for `pesq.update(preds, target)` followed by `pesq.compute()`.

The functional counterpart returns a result [as documented](https://torchmetrics.readthedocs.io/en/stable/audio/perceptual_evaluation_speech_quality.html#torchmetrics.functional.audio.pesq.perceptual_evaluation_speech_quality):
```python
import torch 
from torchmetrics.functional.audio.pesq import perceptual_evaluation_speech_quality as F_PESQ

n = 4
preds = torch.FloatTensor(n, 12345).uniform_(-1, 1)
target = torch.FloatTensor(n, 12345).uniform_(-1, 1)

pesq_score = F_PESQ(preds=preds, target=target, fs=16000, mode="wb")
print("pesq_score:", pesq_score)
#> pesq_score: tensor([1.5882, 1.5080, 1.5149, 1.5997])
```

### Environment

```
Linux
python==3.10.12
torch==2.0.1
torchmetrics== 1.1.2
```

### PR DESCRIPTION
**Pull request by pme0**
## What does this PR do?

Fixes #2095

PESQ ([`torchmetrics.audio.pesq.PerceptualEvaluationSpeechQuality`](https://torchmetrics.readthedocs.io/en/stable/audio/perceptual_evaluation_speech_quality.html#perceptual-evaluation-of-speech-quality-pesq)) docs incorrectly state that a tensor of pesq values is returned. This PR:
- corrects this inconsistency between docs and implementation, namely that `forward` and `compute` return metric value reduced across the batch, instead of a tensor of metrics values for each sample;
- adds a note guiding the user to the functional version of PESQ ([`torchmetrics.functional.audio.pesq.perceptual_evaluation_speech_quality`](https://torchmetrics.readthedocs.io/en/stable/audio/perceptual_evaluation_speech_quality.html#functional-interface)) if a tensor of metrics values for each sample is required;

<details>
  <summary>Before submitting</summary>

- [x] Was this **discussed/agreed** via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/torchmetrics/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure to **update the docs**?
- [ ] Did you write any new **necessary tests**?

</details>

<details>
  <summary>PR review</summary>

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

</details>

----
:books: Documentation preview :books:: https://torchmetrics--2114.org.readthedocs.build/en/2114/

### CODE REVIEW

File: src/torchmetrics/audio/pesq.py [hunk 1]
```diff
@@ -43,13 +43,17 @@ class PerceptualEvaluationSpeechQuality(Metric):
...
 
-    - ``pesq`` (:class:`~torch.Tensor`): float tensor with shape ``(...,)`` of PESQ value per sample
+    - ``pesq`` (:class:`~torch.Tensor`): float tensor of PESQ value reduced across the batch
 
     .. note:: using this metrics requires you to have ``pesq`` install. Either install as ``pip install
         torchmetrics[audio]`` or ``pip install pesq``. ``pesq`` will compile with your currently
         installed version of numpy, meaning that if you upgrade numpy at some point in the future you will
         most likely have to reinstall ``pesq``.
 
+    .. note: the ``forward`` and ``compute`` methods in this class return a single (reduced) PESQ value
```
Comment (Borda) [inline src/torchmetrics/audio/pesq.py:53]:
```suggestion
    .. note:: the ``forward`` and ``compute`` methods in this class return a single (reduced) PESQ value
```

File: src/torchmetrics/audio/pesq.py [hunk 2]
```diff
@@ -43,13 +43,17 @@ class PerceptualEvaluationSpeechQuality(Metric):
...
+        for a batch. To obtain a PESQ value for each sample, you may use the functional counterpart in
+        ``torchmetrics.functional.audio.pesq.perceptual_evaluation_speech_quality()``.
```
Comment (Borda) [inline src/torchmetrics/audio/pesq.py:55]:
```suggestion
        :func:`~torchmetrics.functional.audio.pesq.perceptual_evaluation_speech_quality`.
```

**Review by SkafteNicki — APPROVED** [review]

**Review by Borda — APPROVED** [review]
